test/com/google/enterprise/adaptor/documentum/DocumentumAdaptorTest.java

```diff
@@ -5249,92 +5249,118 @@ private void testModifiedFolderModifiedDocumentsQuery(String query,
         ImmutableMap.of("documentum.modifiedDocumentsQuery", query),
         startPaths(folder),
         new Checkpoint(FEB_1970, DOCUMENT.pad("aaa")),
-        makeExpectedDocIds(folder, expected),
+        expected,
         new Checkpoint(MAR_1970, DOCUMENT.pad("bbb")));
   }
 
   @Test
   public void testModifiedDocumentsQuery() throws Exception {
     String folder = "/FFF1/FFF2";
     String query = "SELECT i_chronicle_id, r_object_id, object_name, "
-        +"r_modify_date, "
+        + "r_modify_date, "
+        + "DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') "
+        + "AS r_modify_date_str FROM dm_sysobject "
+        + "WHERE FOLDER(''/FFF1'',descend) "
+        + "AND ((r_modify_date = DATE(''{0}'',''yyyy-mm-dd hh:mi:ss'') "
+        + "AND r_object_id > ''{1}'') "
+        + "OR (r_modify_date > DATE(''{0}'',''yyyy-mm-dd hh:mi:ss''))) "
+        + "UNION "
+        + "SELECT i_chronicle_id, r_object_id, object_name, r_modify_date, "
         + "DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') "
-        +"AS r_modify_date_str "+"FROM dm_sysobject "
-        +"WHERE FOLDER(''/FFF1'',descend) "
-        +"AND ((r_modify_date = DATE(''{0}'',''yyyy-mm-dd hh:mi:ss'') "
-        +"AND r_object_id > ''{1}'') "
-        +"OR (r_modify_date > DATE(''{0}'',''yyyy-mm-dd hh:mi:ss''))) "
-        +"UNION "
-        +"SELECT i_chronicle_id, r_object_id, object_name, r_modify_date, "
-        +"DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') "
-        +"AS r_modify_date_str "
-        +"FROM dm_sysobject "
-        +"WHERE r_object_id = ''" + DOCUMENT.pad("aaa") + "'' "
-        +"ORDER BY 4 ASC,2 ASC ENABLE(RETURN_TOP 500)";
+        + "AS r_modify_date_str "
+        + "FROM dm_sysobject "
+        + "WHERE r_object_id = ''"
+        + DOCUMENT.pad("aaa")
+        + "'' "
+        + "ORDER BY 4 ASC,2 ASC ENABLE(RETURN_TOP 500)";
 
-    testModifiedFolderModifiedDocumentsQuery(query, "aaa", folder, "bbb");
+    testModifiedDocumentsQuery(query,
+        makeExpectedDocIds(folder, "aaa", folder, "bbb"));
   }
 
   @Test
   public void testModifiedDocumentsQuery_DQLError()
       throws Exception {
     String folder = "/FFF1/FFF2";
+    // select lists do not match and results in dql error. As a result
```

Comment:
On the failures, why not choose much simpler failures, like "SELECT r_modify_date" or "SELECT WHERE" or something that generates an error (both in H2 and iDQL, preferably)? Same for the missing column and IllegalArgumentException, maybe even just "SELECT r_object_id FROM dm_sysobject" and "{0, time}" are enough?